src/MapSearch.js

```diff
@@ -0,0 +1,101 @@
+
+import { useJsApiLoader, GoogleMap, MarkerClusterer, MarkerF, InfoWindow } from '@react-google-maps/api'
+import { useState, useEffect, useCallback } from 'react'
+import axios from 'axios';
+import SaveStation from "./SaveStation"
+import BikeAnimatedSVG from './BikeAnimatedSVG';
+
+
+function MapSearch(url) {
+    const { isLoaded } = useJsApiLoader({
+        googleMapsApiKey: process.env.REACT_APP_GOOGLE_API_KEY
+    })
+
+    const [stationGeo, setStationGeo] = useState()
+    const [activeMarker, setActiveMarker] = useState(null);
+    const [map, setMap] = useState(null);
+    const [mapLoaded, setMapLoaded] = useState(true)
+
+    useEffect(() => {
+        setMapLoaded(false);
+        axios({
+            url: `https://api.citybik.es/${url.data}`,
+            method: "GET",
+        }).then((response) => {
+            const stationData = response.data.network.stations
+            const mappedData = stationData.map((station) => ({ position: { lat: station.latitude, lng: station.longitude }, id: station.id, freeBikes: station.free_bikes, emptySlots: station.empty_slots, stationName: station.name, href: url.data, key: station.id }))
+            setStationGeo(mappedData);
+            setMapLoaded(true)
+        });
+    }, [url.data]);
+
+    useEffect(() => {
+        if (map) {
+            const bounds = new window.google.maps.LatLngBounds();
+            stationGeo.forEach(({ position }) => bounds.extend(position));
+            map.fitBounds(bounds);
+        }
+    }, [map, stationGeo])
+
+    const markerClick = (marker) => {
+        if (marker === activeMarker) {
+            return;
+        }
+        setActiveMarker(marker);
+    }
+
+    const onLoad = useCallback(
+        (map) => {
+            setMap(map);
+        }, []);
+
+    return isLoaded && mapLoaded && stationGeo ? (
+        <GoogleMap
```

Comment:
when i squish the vw, the map gets pushed under the nav :(

Reply:
fixed relevant margin issue. Good catch!